test/e2e/framework/helpers/k8s/k8s_utils.go

```diff
@@ -391,7 +391,13 @@ func (k *KubeCtl) DeleteNamespace(namespace string) error {
 }
 
 func (k *KubeCtl) TearDownNamespace(namespace string) error {
```

Comment:
Should it cleanup both PV and PVC ?

Reply:
Hi @chia7712,
I don't think we should cleanup both PV and PVC in TearDownNamespace.

PV/PVC were only used in "persistent_volume" e2e test.  I think the cleanup responsibility is on the e2e test itself. (Besides, PV is cluster resource and is not bounded to namesapce.)

One of the reason we cleanup Jobs/Pods in TearDownNamespace() is those objects are widely used in serveral e2e tests and sadly we didn't handle all the cleanup.

(IMO, the best practice is each e2e test clean Jobs/Pods by themself. However, it quite difficult to review all the e2e test.)

Reply:
My point was that all e2e should reuse the code to cleanup resources. PV, PVC, pods, and jobs should be cleanup when the e2e is closing even if the e2e does not use such resource. That could make isolation between e2e test and all tests get benefit from it.

Reply:
Agree with Chia-Ping's point. I also believe that we should invoke `DeleteWorkloadAndPods` in the `tearDownNamespace` method.

Reply:
@chenyulin0719 any feedback?

Reply:
@chia7712 
I'm working on extending TearDownNamespace() to cover below namespace level object:
1. Deployment
2. DaemonSet
3. GetReplicaSet
4. GetStatefulSet
5. Pods
6. PVC

>PV, PVC, pods, and jobs should be cleanup when the e2e is closing even if the e2e does not use such resource. 

My previous thought is resource cleanup should be 'explicit'.
ex: "If I create pvc in e2e test, I should delete it after this e2e test.  If I don't use pvc in e2e test,  I don't need to check pvc existing.

However, currently, we're not using the 'explicit' cleanup in every e2e test, and looks like TearDownNamespace() is the shared function to cleanup namespace for multiple e2e test. So I'll follow the suggestion and make it more generic.

@brandboat Thanks for the suggestion. I've checked the DeleteWorkloadAndPods() only deletes one workload object per trigger. (If we have 10 Jobs in namespace, we would need to call DeleteWorkloadAndPods() 10 times). Therefore, I will create a new function for each resource type.

Reply:
@chenyulin0719 thanks for update. sounds good!

Reply:
New commit pushed, 
Also clenaup Deployment,DaemonSet,ReplicaSet,Stateful,PVC in TearDownNamespace()